endpointdata/api.go

```diff
@@ -0,0 +1,62 @@
+// Package endpointdata contains a data structure for collecting data from
+// endpoints. Although scotty uses multiple goroutines to collect data from
+// endpoints, only one goroutine at a time collects data from any given
+// endpoint. A goroutine collecting for an endpoint uses the EndpointData
+// data structure to store state information that needs to persists between
+// collections for that endpoint. Each endpoint gets its own EndpointData
+// data structure which is available during the collection process.
+//
+// The lifespan of an EndpointData instance is as follows. Each endpoint gets
+// the same EndpointData instance for each collection until either the cloud
+// health or cloudwatch configuration changes for that endpoint. When
```

Comment:
As discussed offline, I'm uncomfortable with a generic package having hard-coded explicit references to data sinks. A subsequent refactor seems desirable.